File: ds/stack/mystack.py
class myStack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        if self.is_empty():
            return None
        item = self.items[-1]
        del self.items[-1]
        return item

    def is_empty(self):
        return len(self.items) == 0

s = myStack()
s.push('a')
s.push('b')
s.push('c')

print(s.pop())  # c
print(s.pop())  # b
print(s.pop())  # a

File: ds/stack/MyStack.java
import java.util.ArrayList;

class MyStack {
    private ArrayList<String> items;

    public myStack() {
        items = new ArrayList<>();
    }

    public void push(String item) {
        items.add(item);
    }

    public String pop() {
        if (is_empty()) {
            return null;
        }
        return items.remove(items.size() - 1);
    }

    public boolean is_empty() {
        return items.size() == 0;
    }

    public static void main(String[] args) {
        myStack s = new myStack();
        s.push("a");
        s.push("b");
        s.push("c");

        System.out.println(s.pop());  // c
        System.out.println(s.pop());  // b
        System.out.println(s.pop());  // a
    }
}
